File: Python/Fibonacci.py

# # Program to display the Fibonacci sequence up to n-th term

n = int(input("Please Enter the total number of terms : "))

# first two terms
n1, n2 = 0, 1
temp = 0

# check if the number of terms is valid
if n <= 0:
    print("Please enter a positive integer")
elif n == 1:
    print("Fibonacci sequence upto :",n,":")
    print(n1)
else:
    print("Fibonacci sequence:")
    while temp < n:
        print(n1)
        n3 = n1 + n2
        # Changing values of n1 and n2
        n1 = n2
        n2 = n3
        temp += 1




"""""

Example:
Please Enter the total number of terms : 
11
Fibonacci sequence:
0
1
1
2
3
5
8
13
21
34
55

            ........

Time Complexity :- O(n)
Space Complexity :- O(1)

""""

File: C#/Fibonacci.cs
using System;

/*
 ** Sample I/O
 **
 ** INPUT:
 ** 
 ** "n". Being "n" a positive integer number.
 **
 ** OUTPUT:
 **
 ** Sum till nth term.
 **
 ** Time complexity - O(n)
 ** 
 ** Space complexity - O(1)
 **
 */				
public class Program
{
	private static int Fibonacci(int n)
	{
		var n0 = 0;
		var n1 = 1;
		if(n == 0)
		{
			return 0;
		}
		else if(n == 1)
		{
			return 1;
		}

		var i = 2;
		var result = 1;
		while(i <= n)
		{
			result = n0 + n1;
			n0 = n1;
			n1 = result;
			i++;
		}
		return result;
	}
	
	public static void Main()
	{
		Console.Write("Enter the value of n: ");
		var n = Console.ReadLine();
		int num;
		if(int.TryParse(n, out num))
		{
			var result = Fibonacci(num);
			Console.WriteLine("The result is: " + result);
		}
		else
		{
			Console.WriteLine("n must be a number");
		}
	}
}